packages/client-core/src/common/services/FileThumbnailJobState.tsx

```diff
@@ -197,62 +299,21 @@ export const FileThumbnailJobState = defineState({
       return prev
     })
   },
-  useGenerateThumbnails: async (files: readonly FileBrowserContentType[]) => {
-    const jobState = useMutableState(FileThumbnailJobState)
-    const seenResources = jobState.seenResources
 
-    const fileList = files
-      .map((file) => (file.thumbnailURL || file.type === 'folder' ? undefined : file.key))
-      .filter((key) => key !== undefined)
-      .filter((key) => !seenResources.value.includes(key))
-
-    const query = {
-      key: {
-        $in: fileList
-      },
+  useGenerateThumbnails: async (files: readonly FileBrowserContentType[]) => {
+    useGenerateHelper(files, (file) => (file.thumbnailURL || file.type === 'folder' ? undefined : file.key), {
       thumbnailKey: 'null'
-    } as PaginationQuery
-
-    const resourceQuery = useFind(staticResourcePath, {
-      query: query
     })
-
-    /**
-     * This useEffect will continuously check for new resources that need thumbnails generated until all resources have thumbnails
-     */
-    useEffect(() => {
-      for (const resource of resourceQuery.data) {
-        if (seenResources.value.includes(resource.key)) continue
-        seenResources.merge([resource.key])
-
-        if (resource.type === 'thumbnail') {
-          //set thumbnail's thumbnail as itself
-          API.instance
-            .service(staticResourcePath)
-            .patch(resource.id, { thumbnailKey: resource.key, project: resource.project })
-          continue
-        }
-
-        if (resource.thumbnailKey != null || !extensionCanHaveThumbnail(resource.key.split('.').pop() ?? '')) continue
-
-        const fileJobs = getMutableState(FileThumbnailJobState).jobs
-        if (
-          fileJobs.value.filter((fj) => {
-            fj.key === resource.url
-          }).length < 1
-        ) {
-          fileJobs.merge([
-            {
-              key: resource.url,
-              project: resource.project!
-            }
-          ])
-        }
-      }
-
-      // If there are more files left to be processed in the list we have specified, refetch the query
-      if (resourceQuery.total > resourceQuery.data.length) resourceQuery.refetch()
-    }, [resourceQuery.data])
+  },
+  useGenerateDimensions: async (files: readonly FileBrowserContentType[]) => {
+    useGenerateHelper(
+      files,
+      (file) => (file.type === 'glf' || file.type === 'glb' ? file.key : undefined),
+      {
+        $and: [{ width: '0' }, { height: '0' }, { depth: '0' }]
```

Comment:
If someone uploads a gltf that has no geometry in it (like a .material.gltf or .lookdev.gltf) the system will repeatedly try to calculate its dimensions. We should probably make these fields nullable so that they start as null, then if they are calculated to be 0 we know that we don't need to recalculate them again